socorro/external/es/crashstorage.py

```diff
@@ -124,18 +183,62 @@ def get_index_for_crash(self, crash_date):
 
         return index
 
-    def save_raw_and_processed(self, raw_crash, dumps, processed_crash,
-                               crash_id):
-        """This is the only write mechanism that is actually employed in normal
-        usage.
+    def save_raw_and_processed(self, raw_crash, dumps, processed_crash, crash_id):
+        """Save raw and processed crash data to Elasticsearch
+
+        Note: This is the only write mechanism that is actually employed in
+        normal usage.
+
         """
 
+        # Massage the crash such that the date_processed field is formatted
+        # in the fashion of our established mapping.
+        self.reconstitute_datetimes(processed_crash)
+
+        # Remove bad keys from the raw crash--these keys are essentially
+        # user-provided and can contain junk data
+        remove_bad_keys(raw_crash)
+
+        # Truncate values that are too long
+        truncate_keyword_field_values(FIELDS, raw_crash)
+        truncate_keyword_field_values(FIELDS, processed_crash)
+
+        # Capture crash data size metrics--do this only after we've cleaned up
+        # the crash data
+        self._capture_crash_metrics(raw_crash, processed_crash)
```

Comment:
I moved when these methods/functions get called such that they happen outside the transaction. The transaction shouldn't be affected by any of these failing--it should only wrap issues with indexing in Elasticsearch.